src/core/segment_id.rs

```diff
@@ -46,6 +46,12 @@ impl SegmentId {
         SegmentId(create_uuid())
     }
 
+    /// Creates a new segment id from a uuid string.
+    pub fn from_uuid_string(uuid_string: &str) -> Result<SegmentId, uuid::parser::ParseError> {
```

Comment:
I want to avoid emitting symbols from other libraries (here the `uuid` crate)  to avoid a dependency difficulty in clients.
I created another branch with this part of the problem.